src/Markup/Avalonia.Markup.Xaml.Loader/CompilerExtensions/Visitors/NotSharedVisitor.cs

```diff
@@ -0,0 +1,55 @@
+﻿using XamlX;
+using XamlX.Ast;
+
+namespace Avalonia.Markup.Xaml.XamlIl.CompilerExtensions.Visitors;
+
+class NotSharedVisitor : IXamlAstVisitor
```

Comment:
I'm not sure I understand the purpose of having a visitor here - why isn't only `x:Shared` checked only on the immediate children instead? This is what's done in `XSharedTransformer`. Ideally the logic should be the same (and shared) for two places.